app/src/main/java/org/projectbuendia/client/ui/chart/PatientChartActivity.java

```diff
@@ -351,6 +381,44 @@ public void onClick(View v) {
         });
     }
 
+    /**
+     *  Set the margin_top to a percentage of the height of the root view. Here's why:
+     *  - If we set a layout_weight on the notes panel, we can ensure that the notes panel
+     *    takes up that fraction of the height of its parent.
+     *  - When the keyboard is displayed, however, the panel height adjusts to (weight *
+     *    new parent height).
+     *  - For small values of layout_weight (e.g. < 0.5) this makes the notes panel noticeably
+     *    larger when the keyboard is expanded.
+     *  - So we compute the height at a time when we know the keyboard isn't showing, and assign
+     *    (weight * parent height) to the margin, so that much of the underlying view will always
+     *    be visible.
+     *
+     * @param panelHeightFrac the height to which the notes panel should be set at.
+     */
+    private void setNotesPanelMaxHeightToFracOfParent(final float panelHeightFrac) {
```

Comment:
Yeah, there's a bunch of issues here.
1 - We have to resize the WebView dynamically because the `SlidingUpPanelLayout` doesn't measure the child views when the panel expands and contracts. This is an important performance optimization - measure / layout is expensive.
2 - The IME causes the Window to resize, so if we use a percentage, it's a percentage of the available space when the IME is both open and closed. The actual number of pixels is different though, so you end up with this weird thing where the notes panel resizes by a small amount when the keyboard opens.